### PR TITLE
Add listo and findo commands

### LINKED ISSUE #30: List all orders
As a business owner, I want to list all my orders, so that I can see all my orders at one time.

### LINKED ISSUE #31: Find order(s)
As a business owner, I want to find order(s) using keywords, so I can more efficiently locate details of orders without having to go through the entire list of orders that I am tracking.

### PR DESCRIPTION
**Pull request by janelleljt**
In this PR, I've made changes to the LogicManager and ModelManager in order to display a filtered list. I also updated the filtering logic to find orders which correspond to the keyword.

### CODE REVIEW

File: src/main/java/tracko/commons/core/Messages.java [hunk 1]
```diff
@@ -8,6 +8,7 @@ public class Messages {
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
-    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    public static final String MESSAGE_ORDERS_FOUND_OVERVIEW = "%1$d orders found!";
+    public static final String MESSAGE_ONE_ORDER_FOUND = "1 order found!";
```
Comment (donljh) [inline src/main/java/tracko/commons/core/Messages.java]: Can rename to just MESSAGE_ORDERS_FOUND = "%1$d order(s) found!", and change the logic as such to make the general code shorter and easier to test.

File: src/main/java/tracko/logic/commands/order/FindOrderCommand.java [hunk 2]
```diff
@@ -1,36 +1,43 @@
...
 import tracko.logic.commands.Command;
 import tracko.logic.commands.CommandResult;
 import tracko.model.Model;
-import tracko.model.order.NameContainsKeywordsPredicate;
+import tracko.model.order.OrderContainsKeywordsPredicate;
 
 /**
  * Finds and lists all persons in address book whose name contains any of the argument keywords.
  * Keyword matching is case insensitive.
  */
```
Comment (donljh) [inline src/main/java/tracko/logic/commands/order/FindOrderCommand.java]: I think you can change the JavaDoc comments here to fit the application context

File: src/main/java/tracko/logic/commands/order/FindOrderCommand.java [hunk 3]
```diff
@@ -1,36 +1,43 @@
...
-        return null;
+        requireNonNull(model);
+        model.updateFilteredOrderList(predicate);
+
+        if (model.getFilteredOrderList().size() == 1) {
+            return new CommandResult(Messages.MESSAGE_ONE_ORDER_FOUND);
+        }
+
+        return new CommandResult(
+                String.format(Messages.MESSAGE_ORDERS_FOUND_OVERVIEW, model.getFilteredOrderList().size()));
```
Comment (donljh) [inline src/main/java/tracko/logic/commands/order/FindOrderCommand.java]: As mentioned above, can change the orders message to reduce number of lines of code here

File: src/main/java/tracko/model/ModelManager.java [hunk 4]
```diff
@@ -96,6 +100,23 @@ public ObservableList<Order> getOrderList() {
         return trackO.getOrderList();
     }
 
+    //=========== Filtered Person List Accessors =============================================================
+
+    /**
+     * Returns an unmodifiable view of the list of {@code Order} backed by the internal list of
+     * {@code versionedAddressBook}
+     */
```
Comment (donljh) [inline src/main/java/tracko/model/ModelManager.java]: Can possibly change the comments here to fit the application

File: src/test/java/tracko/model/person/NameContainsKeywordsPredicateTest.java [hunk 5]
```diff
@@ -1,76 +1,78 @@
...
+//
+//        // Non-matching keyword
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("Carol"));
+//        assertFalse(predicate.test(new OrderBuilder().withName("Alice Bob").build()));
+//
+//        // Keywords match phone, email and address, but does not match name
+//        predicate = new NameContainsKeywordsPredicate(Arrays.asList("12345", "alice@email.com", "Main", "Street"));
+//        assertFalse(predicate.test(new OrderBuilder().withName("Alice").withPhone("12345")
+//                .withEmail("alice@email.com").withAddress("Main Street").build()));
+//    }
```
Comment (donljh) [inline src/test/java/tracko/model/person/NameContainsKeywordsPredicateTest.java]: Can maybe look into re-implementing these test cases somehow. I think a lot of them can be reimplemented by changing a few lines here and there.

**Review by donljh — CHANGES_REQUESTED** [review]

File: src/test/java/tracko/model/person/NameContainsKeywordsPredicateTest.java (hunk 5 above)
Comment (janelleljt) [reply src/test/java/tracko/model/person/NameContainsKeywordsPredicateTest.java]: yup, reimplemented the tests

File: src/main/java/tracko/logic/commands/order/FindOrderCommand.java (hunk 3 above)
Comment (janelleljt) [reply src/main/java/tracko/logic/commands/order/FindOrderCommand.java]: Changed!

**Review by sohenze — APPROVED** [review]
Looks good!

**Review by donljh — APPROVED** [review]

**Pull request merged**